include/boost/geometry/algorithms/detail/distance/geometry_to_segment_or_box.hpp

```diff
@@ -316,6 +345,48 @@ class geometry_to_segment_or_box
                     Strategy
                 >::apply(*it_min, *sit_min, strategy);
         }
+
+        bool is_cartesian = boost::is_same
+                <
+                    typename cs_tag<Geometry>::type,
+                    cartesian_tag
+                >::type::value;
+
+
+        bool is_polygon = boost::is_same
+                <
+                    typename tag<Geometry>::type,
+                    polygon_tag
+                >::type::value;
+
+        bool is_ring = boost::is_same
+                <
+                    typename tag<Geometry>::type,
+                    ring_tag
+                >::type::value;
+
+
+        bool is_mpolygon = boost::is_same
+                <
+                    typename tag<Geometry>::type,
+                    multi_polygon_tag
+                >::type::value;
+
+        bool is_box = boost::is_same
+                <
+                    typename tag<SegmentOrBox>::type,
+                    box_tag
+                >::type::value;
+
+        if ((is_polygon || is_mpolygon) && is_cartesian && is_box)
+        {
+            strategy::distance::services::swap<Strategy>::apply(res);
```

Comment:
What does it do? What does it swap with what? Typically a function like that would swap two things with each other. Furthermore, is `services` namespace really a good place for it? Isn't it a detail?

Reply:
Is this function here because `res` may either be a number of an "info" (distance + 2 points)? If that's the case then it doesn't swap anything if it is a number right?

Reply:
yes, and it actually swaps the two points in the case of closest_points and nothing in case of distance. I renamed it to `swap_result_points`. I placed it into `services` namespace since it is a result manipulation function such as `result_from_distance`